test/unit/BPool.t.sol

```diff
@@ -881,29 +881,170 @@ contract BPool_Unit_Rebind is BasePoolTest {
+  function test_HappyPath(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+  }
 
-  function test_Unset_Record() private view {}
+  function test_Revert_NotController(
+    address _controller,
+    address _caller,
+    Unbind_FuzzScenario memory _fuzz
+  ) public happyPath(_fuzz) {
+    vm.assume(_controller != _caller);
+    bPool.set__controller(_controller);
 
-  function test_Push_UnderlyingBalance() private view {}
+    vm.prank(_caller);
+    vm.expectRevert('ERR_NOT_CONTROLLER');
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+  }
 
-  function test_Push_UnderlyingFee() private view {}
+  function test_Revert_NotBound(address _token, Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    for (uint256 i = 0; i < _fuzz.previousTokensAmount; i++) {
+      vm.assume(_token != _fuzz.previousTokens[i]);
+    }
 
-  function test_Emit_LogCall() private view {}
+    vm.expectRevert('ERR_NOT_BOUND');
+    bPool.unbind(_token);
+  }
+
+  function test_Revert_Finalized(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setFinalize(true);
+
+    vm.expectRevert('ERR_IS_FINALIZED');
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+  }
+
+  function test_Revert_Reentrancy(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    // Assert that the contract is accessible
+    assertEq(bPool.call__mutex(), false);
+
+    // Simulate ongoing call to the contract
+    bPool.set__mutex(true);
+
+    vm.expectRevert('ERR_REENTRY');
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+  }
+
+  function test_Set_TotalWeight(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+
+    assertEq(bPool.call__totalWeight(), _fuzz.totalWeight - _fuzz.denorm);
+  }
+
+  function test_Set_TokenArray(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    address _lastTokenBefore = bPool.call__tokens()[bPool.call__tokens().length - 1];
+
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+
+    // Only check if the token is not the last of the array (that item is always poped out)
+    if (_fuzz.tokenIndex != _fuzz.previousTokensAmount - 1) {
+      address _tokenToUnbindAfter = bPool.call__tokens()[_fuzz.tokenIndex];
+      assertEq(_tokenToUnbindAfter, _lastTokenBefore);
+    }
+  }
+
+  function test_Set_Index(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    address _lastTokenBefore = bPool.call__tokens()[bPool.call__tokens().length - 1];
+
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
+
+    // Only check if the token is not the last of the array (that item is always poped out)
+    if (_fuzz.tokenIndex != _fuzz.previousTokensAmount - 1) {
+      assertEq(bPool.call__records(_lastTokenBefore).index, _fuzz.tokenIndex);
+    }
+  }
+
+  function test_Unset_TokenArray(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
```

Comment:
Similar reasoning to the previous comment—`Pop` would be explicit, and `Push` in the opposite operation.